sympy/simplify/simplify.py

```diff
@@ -931,6 +931,51 @@ def _ratsimpmodprime(a, b, allsol, N=0, D=0):
     return (c*r.q)/(d*r.p)
 
 
+def rootofsimp(expr):
+    """
+    Reduce expressions containing ``RootOf`` types modulo their defining
+    polynomials.
+
+    A root `r` of a polynomial `p(x) = a_n x^n + ... + a_0` satisfies
+    `p(r) = 0` by definition. Therefore, any power of `r` greater than
+    `n` can be rewritten in terms of smaller powers. In general, any
+    polynomial or rational expression of `r` can be reduced modulo its
+    defining polynomial `p`.
+
+    ``rootofsimp`` rewrites the expression as a numerator-denominator
+    pair, computes the modular inverse of the denominator, and returns
+    the product of this inverse with the numerator modulo the defining
+    polynomial of the root. This is done for each ``RootOf`` appearing
+    in the expression.
+
+    Examples
+    ========
+
+    >>> from sympy import rootofsimp, RootOf
+    >>> from sympy.abc import x,y
+    >>> r = RootOf(x**4 - x - 1, 0)
+    >>> rootofsimp(r**4)
+    RootOf(x**4 - x - 1, 0) + 1
+    >>> rootofsimp(r**8)
+    2*RootOf(x**4 - x - 1, 0) + RootOf(x**4 - x - 1, 0)**2 + 1
+    >>> rootofsimp(1/r)
+    -1 + RootOf(x**4 - x - 1, 0)**3
+
+    """
+    from sympy.polys.rootoftools import RootOf
+    if not expr.has(RootOf):
+        return expr
+
+    for root in expr.find(RootOf):
+        # rewrite the defining polynomial in terms of the root, itself
+        modulus = root.poly.xreplace({root.poly.gen:root})
+
+        numer,denom = cancel(expr).as_numer_denom()
+        denom = denom.as_poly(root).invert(modulus)
```

Comment:
That's a good question. The simplest solution would be to raise a divide by zero error. Since `cancel` is used we shouldn't expect `denom` to share any factors with `numer`, so if zero modulo the defining polynomial is really in the denominator then we're dividing by zero.

In fact, this is the behavior of Maple.

```maple
a := RootOf(z**3 - z + 1);
                              /  3         \
                   a := RootOf\_Z  - _Z + 1/
expr := 1/(a**3 - a + 1);
                                   1                        
    expr := ------------------------------------------------
                                3                           
                  /  3         \          /  3         \    
            RootOf\_Z  - _Z + 1/  - RootOf\_Z  - _Z + 1/ + 1
simplify(expr);
Error, (in simplify/RootOf) numeric exception: division by zero
```

and in Sage

```python
sage: R.<x> = QQ['x']
sage: f = x**3 - x + 1
sage: I = R.ideal(f)
sage: Q.<a> = R.quotient(I)
sage: expr = 1/(a**3 - a + 1)
Traceback (most recent call last):
...
ZeroDivisionError: element 0 of quotient polynomial ring not invertible
```

so to me I think that's argument enough for a divide by zero error. But instead of just letting `Poly.invert()` raise the error I'll catch it and provide a more descriptive message.